pydantic_factories/factory.py

```diff
@@ -485,9 +496,11 @@ def build(cls, factory_use_construct: bool = False, **kwargs: Any) -> T:
 
         Note - this is supported only for pydantic models
         """
-        for field_name, model_field in cls.get_model_fields(cls._get_model()):
-            if model_field.alias:
+        model = cls._get_model()
+        for field_name, model_field in cls.get_model_fields(model):
+            if cls._should_use_alias_name(model_field, model):
```

Comment:
I guess we are off what we intended to do, we agreed to allow both `field_name` and `alias` to be accepted as the key for the field i.e. if we passed `aliased_field="one"` or `special_field="one"` to the `build` method, it should be accepted.

Reply:
I agree, please update the implementation to mirror the discussion in the ticket.

Reply:
I think I start to see what are you talking about. For pydantic models with the flag in the config, the method always returns `False`.  

And it still works, cause we don't need to change a field name, both of them (alias and field name) is acceptable.